lib/src/sprite_animation.dart

```diff
@@ -66,21 +66,24 @@ class SpriteAnimationData {
   }
 
   /// Works just like [SpriteAnimationData.variable] but uses the same [stepTime] for all frames
-  SpriteAnimationData.sequenced({
+  factory SpriteAnimationData.sequenced({
     @required int amount,
     @required double stepTime,
     @required Vector2 textureSize,
     int amountPerRow,
     Vector2 texturePosition,
     bool loop = true,
-  }) : this.variable(
-          amount: amount,
-          amountPerRow: amountPerRow,
-          texturePosition: texturePosition,
-          textureSize: textureSize,
-          loop: loop,
-          stepTimes: stepTime != null ? List.filled(amount, stepTime) : null,
-        );
+  }) {
+    assert(stepTime != null);
```

Comment:
Even better!

Reply:
technically we could check every element of the list in the other constructor (but that is unrelated to this change as if you come through this it will be impossible some are null and some are not).

lmk if you like that or if it's overkill.